docs/models_dev_guide.md

```diff
@@ -0,0 +1,263 @@
+# Models Developer's Guide
+
+This document describes the design of the model management system in ETA and
+best practices for using it.
+
+
+## Motivation
+
+The ETA library relies on a number of large models in order to power its
+analytics. Models are trained on different machines, by different developers,
+and constantly need to be shared. We frequently generate new versions of a
+given model over time as we improve the implementation of the analytic and
+train it on evolving datasets. In addition, models are valuable, so they need
+to be archived and backed up to ensure that they remain secure and readily
+accessible.
+
+The ETA library provides a simple, yet powerful interface that satisifies
+these constraints.
+
+
+## Design
+
+The `eta.core.models` module implements the core interface for managing models
+in ETA. The underlying design principle of the interface is that all models
+are stored in cloud storage and can be easily downloaded to a local ETA
+installation as needed using a simple version-aware naming system.
+
+### Model versions
+
+Each model is identified by a unique string of the form `<name>@<version>`,
+where `<name>` is the canonical name of the model and `<version>` is an
+optional version of the model.
+
+If a model is registered without a version, it can be referenced simply by
+its `<name>`. If a model is registered with a version, it can be referenced
+explicitly using the `<name>@<version>` syntax. The `<name>` syntax can
+also be used with versioned models, in which case the latest version of the
+model (i.e. the largest version number) will be referenced.
+
+#### Models search path
+
+The `models_dirs` field of the ETA config defines the list of directories to
+use when performing all models-related operations. If you add a new models
+directory to your path, you can use the `eta.core.models.init_models_dir`
+function to initialize it for use with the model management interface.
```

Comment:
This here probably needs a bit more explanation.  When would you need to?  I am guessing only if it is added mid-execution, otherwise it will get loaded/initialized during startup?